clients/drcachesim/tracer/tracer.cpp

```diff
@@ -477,6 +539,7 @@ memtrace(void *drcontext, bool skip_size_cap)
         if (is_num_refs_beyond_global_max()) {
```

Comment:
Maybe add a test that combines the new options with the max global/thread refs options.

Reply:
These max options operate quite differently: they only count at output time and do not affect instru, so there should be no interoperability concerns.  Manual testing shows them mixing just fine.  It's hard to do an automated test w/o making it flaky b/c of the imprecise values: can't easily have the output regex ensure a precise enough value.

There is a todo item (added comment in this file somewhere too) to reimplement the global max using drbbdup to switch to non-tracing instead of today where it keeps tracing but throws away the data.